swap/src/protocol/alice/execution_setup.rs

```diff
@@ -39,7 +39,7 @@ impl Default for Behaviour {
 }
 
 impl Behaviour {
-    fn run(&mut self, bob: PeerId, state0: State0) {
+    pub fn run(&mut self, bob: PeerId, state0: State0) {
```

Comment:
I had a comment in the previous commit about how you are getting away with this being private :grin: 

It is easier to review if code is used right away and you are not committing dead code separately (does the compiler not emit a dead-code warning for the other commit?). I would probably squash this one together with the one where you introduce it, even if it makes it slightly bigger.

Reply:
Both commits are huge in this case. I thought it would be better for the reviewers to have them separate.

Reply:
I'd take a bigger but coherent diff over one that introduces code that is not used. It is harder to give feedback if I have to navigate back and forth between separate commits to understand the bigger picture.